python/pyspark/sql/udf.py

```diff
@@ -131,58 +130,24 @@ def _create_py_udf(
     else:
         is_arrow_enabled = useArrow
 
-    regular_udf = _create_udf(f, returnType, PythonEvalType.SQL_BATCHED_UDF)
-    try:
-        is_func_with_args = len(getfullargspec(f).args) > 0
-    except TypeError:
-        is_func_with_args = False
+    eval_type: int = PythonEvalType.SQL_BATCHED_UDF
+
     if is_arrow_enabled:
+        try:
+            is_func_with_args = len(getfullargspec(f).args) > 0
+        except TypeError:
+            is_func_with_args = False
         if is_func_with_args:
-            return _create_arrow_py_udf(regular_udf)
+            require_minimum_pandas_version()
+            require_minimum_pyarrow_version()
```

Comment:
@xinrong-meng Do we want to fail here? or fallback to non-Arrow similar to Python UDTF?

https://github.com/apache/spark/blob/5526efc206e7f98f73a141151477a5022efff05e/python/pyspark/sql/udtf.py#L116-L127

cc @allisonwang-db

Reply:
I would suggest for Arrow Python UDF, we fail in Spark 3.5. And when it's turned on by default in Spark 4.0, we fall back. WDYT?

Reply:
Sure, then I'll just keep it as-is. Thanks.